.stylelintrc.yaml

```diff
@@ -1,5 +1,13 @@
-extends: stylelint-config-standard-scss
```

Comment:
> The postcss-styled-syntax package for Stylelint is able to parse CSS-in-JS in .tsx files, but does not catch issues in template literals such as the !default hangover from Sass. Emotion [is using Stylis v4](https://emotion.sh/docs/emotion-11#stylis-v4) to process styles, though there are no vendor Stylelint plugins for Stylis.

I think this depends on the ruleset enabled.

> [postcss-styled-syntax](https://www.npmjs.com/package/postcss-styled-syntax) is not able to correctly parse function interpolations in template literal expressions in conjunction with the stylelint-config-standard-scss rules, so errors on e.g. color: ${({ theme }) => theme.custom.colorBlue5};. These rules are now targeting .scss files specifically.

Good point. Looking at comments in the `postcss-styled-syntax` repo, I think the SCSS ruleset is just a poor fit for CSS-in-JS. But the `stylelint-config-standard` ruleset is, I think, a good fit. 

Using 
```yaml
extends:
  - stylelint-config-standard
```
stylelint seems to be catching emotion errors much better. Adding a `min-height: 55pxx;` and `display: flex !default;` to test,
```
➤ YN0000: [mit-open]:
➤ YN0000: [mit-open]: src/page-components/Header/Header.tsx
➤ YN0000: [mit-open]:  10:17  ✖  Unexpected unknown unit "pxx"             unit-no-unknown
➤ YN0000: [mit-open]:  11:17  ✖  Unexpected unknown annotation "!default"  annotation-no-unknown
➤ YN0000: [mit-open]:
➤ YN0000: [mit-open]: 2 problems (2 errors, 0 warnings)
➤ YN0000: [mit-open]:
➤ YN0000: [mit-open]: Process exited (exit code 2), completed in 3s 261ms
```

Which seems great!

**Suggestions:**
- Let's use `stylelint-config-standard` as the base
- We can add a `.stylelintignore` at root, and maybe ignore `/vendor/**/*.css` (or maybe `.vendor.css?`), and move the ckeditor file to there. I agree...Not sure why it wasn't erroring before. (Was it css? Maybe we only linted scss before.)
    -  might also want to ignore `build/` code. I had a build/ directory lying around from when I did a prod build a some point, and ended up with 1000 css errors...

> Emotion [best practices](https://emotion.sh/docs/best-practices#use-typescript-and-object-styles) prefer object styles over template literals...

I dunno, I find their reasoning unconvincing. They say you don't get intellisense or typechecking. But it seems to me that, when it comes to CSS, `stylelint` is much better than typechecking. And I do get intellisense...

<img width="400" alt="Screenshot 2024-01-18 at 2 57 12 PM" src="https://github.com/mitodl/mit-open/assets/9010790/389c61ab-dc44-4b92-8fca-8d77fa163afa">

Reply:
Re intellisense... it turns out I have https://marketplace.visualstudio.com/items?itemName=styled-components.vscode-styled-components installed, which is what's giving the highlighting and intellisense.

Reply:
Agree with all - nice to catch that `!default`; pushed https://github.com/mitodl/mit-open/pull/390/commits/45dd27a1d